python/lsst/rubintv/production/utils.py

```diff
@@ -258,3 +262,63 @@ def remakeDay(channel, dayObs, remakeExisting=False, notebook=True, logger=None,
         tvChannel.callback(dataId)
         if channel in ['auxtel_metadata']:
             sleep(1.5)  # metadata creation is fast enough that we hit the GCS rate limit without a pause
+
+
+def pushTestImageToCurrent(channel, duration=15):
+    """Push a test image to a channel to see if it shows up automatically.
+
+    Leaves the test image in the bucket for ``duration`` seconds and then
+    removes it. ``duration`` cannot be more than 60s, as test images should not
+    be left in the bucket for long, and a minute should easily be long enough
+    to see if things are working.
+
+    NB: this function is designed for interactive use in notebooks and blocks
+    for ``duration`` and then deletes the file.
+
+    Parameters
+    ----------
+    channel : `str`
+        The name of the lsst.rubintv.production channel. The actual channel
+        object is created internally.
+    duration : `float`, optional
+        The duration to leave the test image up for, in seconds.
+    """
+    logger = logging.getLogger(__name__)
+
+    from google.cloud import storage
+
+    if channel not in CHANNELS:
+        raise ValueError(f"Channel {channel} not in {CHANNELS}")
+    if channel in ['auxtel_metadata']:
+        raise ValueError(f'Pushing test data not supported for {channel}')
+    if duration > 60:
+        raise ValueError(f'Maximum time to leave test images in buckets is 60s, got {duration}')
+
+    client = storage.Client()
+    bucket = client.get_bucket('rubintv_data')
+    prefix = f'{channel}/{PREFIXES[channel]}'
+    blobs = list(bucket.list_blobs(prefix=prefix))
+
+    logger.info(f'Found {len(blobs)} for channel {channel} in bucket')
+
+    # names are like
+    # 'auxtel_monitor/auxtel-monitor_dayObs_2021-07-06_seqNum_100.png'
+    days = set([b.name.split(f'{prefix}_dayObs_')[1].split('_seqNum')[0] for b in blobs])
+    days = [int(d.replace('-', '')) for d in days]  # days are like 2022-01-02
+    recentDay = max(days)
+
+    seqNums = getPlotSeqNumsForDayObs(channel, recentDay, bucket)
+    newSeqNum = max(seqNums) + 1
```

Comment:
I'm a bit worried that `+ 1` may allow a race condition between this test image and potential new incoming images.  If everything blocks, so no update can happen, then this is fine.  If this is something that may prevent an image being generated with that `newSeqNum` value, and image generation will resume after this test file is removed (the catchup code for example), then it's also fine.

Reply:
Oh no, you're 100% right about the race conditions, and overwriting images etc. This code is absolutely *only* for use when things aren't online and taking data! When things are rolling in then we have images flowing to test the frontend code, and wouldn't need to use this. This is entirely for developer use to test when we don't have any real data flowing.

Reply:
Given the utility functions that now exist for regenerating whole days etc, plus the catchup services, and the intention that this is only used by experts, I personally think that this is all fine though. If you disagree please let me know though.